### DISCUSSION TITLE
CPU cores/threads used

### DISCUSSION THREAD

**Discussion by NunoMac** [ANTsX/ANTs#1709] [Q&A]
Hi, I m running ANTs on a server. My question is: will it run using the maximum number of threads available? Or is it a good idea to define that manually?

Right now, i am applying the following to do the registration:
        -d 3 \
	--float 0 \
	-v 1 \
	-o [${output},${output}.nii.gz] \
	--interpolation WelchWindowedSinc \
	--use-histogram-matching ${histomatching} \
	--winsorize-image-intensities [0.005,0.995] \
	--initial-moving-transform [${fixed},${moving},0] \
	-t rigid[${lstep}] \
	-m MI[${fixed},${moving},1,32,Regular,0.25] \
	-c [200x200x200x0,${convlin},10] \
	--shrink-factors 8x4x2x1 \
	--smoothing-sigmas 3x2x1x0vox \
	 -t Affine[${lstep}] \
	 -m MI[${fixed},${moving},1,32,Regular,0.25] \
	 -c [200x200x200x0,${convlin},10] \
	 --shrink-factors 8x4x2x1 \
	 --smoothing-sigmas 3x2x1x0vox \
	 -t SyN[${synstep},${synpenalty1},${synpenalty2}] \
	 -m CC[${fixed},${moving},1,${ccradius}] \
	 -c [200x200x100x50,${convsyn},10] \
	 --shrink-factors 8x4x2x2 \
	 --smoothing-sigmas 4x3x2x1vox \

should I add
'-n 8' at the begining to force the use of 8 threads?

I haven't found this being used in other scripts...

Thanks!

Nuno

**Comment by cookpa** [comment]
Multi-threading in ITK filters is controlled by the environment variable `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS`. Some of the ANTs scripts allow this to be set dynamically with `-n`, but if you are running the `antsRegistration` binary directly, you'll want to set the variable.

**Reply by NunoMac** [reply]
Thank you for the tips! I set it to run on the server environment as `environment = "ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=8"` (I read that the gains above 8 are probabyl not worth it) and will check if it worked and improved processing time.
